prefect_azure/container_instance.py

```diff
@@ -0,0 +1,694 @@
+
+    type: Literal["container-instance-job"] = Field(
+        default="container-instance-job", description="The slug for this task type."
+    )
+    aci_credentials: ContainerInstanceCredentials
+    resource_group_name: str = Field(
+        default=...,
+        title="Azure Resource Group Name",
+        description=(
+            "The name of the Azure Resource Group in which to run Prefect ACI tasks."
+        ),
+    )
+    subscription_id: SecretStr = Field(
+        default=...,
+        title="Azure Subscription ID",
+        description="The ID of the Azure subscription to create containers under.",
+    )
+    image: Optional[str] = Field(
+        default_factory=get_prefect_image_name,
+        description=(
+            "The image to use for the Prefect container in the task. This value "
+            "defaults to a Prefect base image matching your local versions."
+        ),
+    )
+    entrypoint: Optional[str] = Field(
+        default=DEFAULT_CONTAINER_ENTRYPOINT,
+        description=(
+            "The entrypoint of the container you wish you run. This value "
+            "defaults to the entrypoint used by Prefect images and should only be "
+            "changed when using a custom image that is not based on an official "
+            "Prefect image. Any commands set on deployments will be passed "
+            "to the entrypoint as parameters."
+        ),
+    )
+    image_registry: Optional[prefect.infrastructure.docker.DockerRegistry] = None
+    cpu: float = Field(
+        title="CPU",
+        default=ACI_DEFAULT_CPU,
+        description=(
+            "The number of virtual CPUs to assign to the task container. "
+            f"If not provided, a default value of {ACI_DEFAULT_CPU} will be used."
+        ),
+    )
+    gpu_count: Optional[int] = Field(
+        title="GPU Count",
+        default=None,
+        description=(
+            "The number of GPUs to assign to the task container. "
+            "If not provided, no GPU will be used."
+        ),
+    )
+    gpu_sku: Optional[str] = Field(
+        title="GPU SKU",
+        default=None,
+        description=(
+            "The Azure GPU SKU to use. See the ACI documentation for a list of "
+            "GPU SKUs available in each Azure region."
+        ),
+    )
+    memory: float = Field(
+        default=ACI_DEFAULT_MEMORY,
+        description=(
+            "The amount of memory in gigabytes to provide to the ACI task. Valid "
+            "amounts are specified in the Azure documentation. If not provided, a "
+            f"default value of  {ACI_DEFAULT_MEMORY} will be used unless present "
+            "on the task definition."
+        ),
+    )
+    subnet_ids: Optional[List[str]] = Field(
+        default=None,
+        description="A list of Azure subnet IDs the container should be connected to.",
+    )
```

Comment:
Small nit for consistent ID casing.
```suggestion
    subnet_ids: Optional[List[str]] = Field(
        default=None,
        title="Subnet IDs",
        description="A list of Azure subnet IDs the container should be connected to.",
    )
```